src/dft/backends/mklgpu/CMakeLists.txt

```diff
@@ -32,12 +32,14 @@ add_library(${LIB_OBJ} OBJECT
 )
 add_dependencies(onemkl_backend_libs_dft ${LIB_NAME})
 
-target_include_directories(${LIB_OBJ}
-  PUBLIC ${ONEMKL_INTERFACE_INCLUDE_DIRS}
-)
 target_include_directories(${LIB_NAME}
   PUBLIC ${ONEMKL_INTERFACE_INCLUDE_DIRS}
 )
+
+target_compile_options(${LIB_OBJ}
+  BEFORE PRIVATE -iquote $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
```

Comment:
This deserves a comment to explain the header conflict and why the `iquote` is needed.
That looks better than I expected, this is great!

Reply:
Added comment in [562f1db](https://github.com/oneapi-src/oneMKL/pull/575/commits/562f1dbdc763fd65107eb3247b30fdfc053189b0)